bin/add_sh_to_taxonomy.py

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env python3
+#@author Jeanette Tångrot
+
+# Adds SH information to ASV table based on vsearch usearch_global results in blast6 format.
+#
+# Usage: add_sh_to_taxonomy.py <seq2sh.tsv> <SHs.tax> <tax.tsv> <blastout.tab> <outfile> [--species]
+#   Input:
+#          <seq2sh.tsv>   : List with sequence to SH matchings, SH level 1.5, for each sequence in
+#                           the database used by vsearch. Each row should contain database sequence
+#                           name, SH name, and taxon number for the SH, separated by tabs.
+#          <SHs.tax>      : List of complete taxonomy for each SH.
+#          <tax.tsv>      : ASV taxonomy table to add SH assignments to
+#          <blastout.tab> : Results from vsearch usearch_global in blast6 format
+#          <outfile>      : Name of results file, i.e. the updated ASV taxonomy table
+#   Options:
+#          --species : Whether or not to include species information. Default: false
+
+import sys
+import pandas as pd
+
+with_species = False
+
+# Argument check
+if len(sys.argv) != 6 and len(sys.argv) != 7:
+    exit("Usage: add_sh_to_taxonomy.py <seq2sh.tsv> <SHs.tax> <tax.tsv> <blastout.tab> <outfile> [--species]")
+if len(sys.argv) == 7:
+    if sys.argv[6] == '--species':
+        with_species = True
+    else:
+        exit("Usage: add_sh_to_taxonomy.py <seq2sh.tsv> <SHs.tax> <tax.tsv> <blastout.tab> <outfile> [--species]")
+outfile = sys.argv[5]
+
+# Set number of ranks - include species if --species flag is used
+if with_species:
+    num_ranks = 8
+    tax_entries = ['Domain','Kingdom','Phylum','Class','Order','Family','Genus','Species','SH','confidence']
+else:
+    num_ranks = 7
+    tax_entries = ['Domain','Kingdom','Phylum','Class','Order','Family','Genus','SH','confidence']
```

Comment:
I think we might be forced to be more flexible with ranks, both which and how many, at some point -- could this be read from the `tax.tsv` file header, to make a transition easier? In essence, what you would need in insert an `SH` column as the next to last and `num_ranks` could be counted from the header.

Reply:
That's a great suggestion, thanks! I don't know why I didn't do it like that to start with... That also removes the need to check --species flag.